include/boost/fusion/container/list/list.hpp

```diff
@@ -13,6 +13,95 @@
 ///////////////////////////////////////////////////////////////////////////////
 // Without variadics, we will use the PP version
 ///////////////////////////////////////////////////////////////////////////////
+#if !defined(BOOST_FUSION_HAS_VARIADIC_LIST)
 # include <boost/fusion/container/list/detail/cpp03/list.hpp>
+#else
 
+///////////////////////////////////////////////////////////////////////////////
+// C++11 interface
+///////////////////////////////////////////////////////////////////////////////
+#include <boost/fusion/container/list/detail/list_to_cons.hpp>
+
+namespace boost { namespace fusion
+{
+    struct nil_;
+    struct void_;
+
+    template <>
+    struct list<>
+        : detail::list_to_cons<>::type
+    {
+    private:
+        typedef detail::list_to_cons<> list_to_cons;
+        typedef list_to_cons::type inherited_type;
+
+    public:
+        BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+        list()
+            : inherited_type() {}
+
+        template <typename Sequence>
+        BOOST_FUSION_GPU_ENABLED
+        list(Sequence const& rhs)
+            : inherited_type(rhs) {}
+
```

Comment:
@djowel , I added move ctor/assgin for C++11 interface https://github.com/Flast/fusion/commit/d5ce74dffd0d461a931dc90cb36b5919ba80f99b , but not for C++03 and cons because modifying cons is a deal (the ctor of cons requires overall reimplementing).
Is this OK in this PR?